core/spec/models/spree/order_updater_spec.rb

```diff
@@ -279,5 +279,71 @@ module Spree
         updater.update
       end
     end
+
+    describe "#persist_totals" do
+      let!(:order) { super() }
+      let(:updated_at) { Time.now }
+      let(:validation_order) { mock_model(Spree::Order, valid?: true) }
+      let(:totals) { attributes.merge(updated_at: updated_at) }
+
+      let(:attributes) do
+        {
+          payment_state:        nil,
+          shipment_state:       nil,
+          item_count:           0,
+          item_total:           0,
+          adjustment_total:     0,
+          included_tax_total:   0,
+          additional_tax_total: 0,
+          payment_total:        0,
+          shipment_total:       0,
+          promo_total:          0,
+          total:                0
+        }
+      end
+
+      before do
+        allow(Time).to receive(:now).and_return(updated_at)
+        allow(Spree::Order).to receive(:new).with(attributes).
+          and_return(validation_order)
+      end
+
+      context 'when the columns are updated' do
+        before do
+          allow(order).to receive(:update_columns).and_return(true)
+        end
+
+        it 'persists the expected totals' do
+          expect(updater.persist_totals).to be(true)
+          expect(order).to have_received(:update_columns).with(totals)
+        end
+
+        it 'returns true' do
+          expect(updater.persist_totals).to be(true)
+        end
+      end
+
+      context 'when the columns are not updated' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.